tasks/kernel_matrix_testing/download.py

```diff
@@ -72,12 +61,12 @@ def download_rootfs(ctx, rootfs_dir, vmconfig_template):
         if not os.path.exists(path):
             to_download.append(f)
 
-    disks_to_download = list()
+    disks_to_download: list[str] = list()
     for vmset in vmconfig_template["vmsets"]:
-        if vmset["arch"] != arch:
+        if vmset.get("arch") != arch:
```

Comment:
Isn't brackets better here? If `arch` is not present that means we have an invalid configuration which should cause an exception.

Reply:
Because this type is configured with `total=False` (signals that some keys might not be present), this gets flagged because we haven't checked that `arch` exists in the dictionary. I could put a check above to raise an exception if the key doesn't exist.

Reply:
Ah got it. Yeah a check would be useful IMO.